x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/expandable_section/expandable_section_results.tsx

```diff
@@ -127,6 +186,99 @@ export const ExpandableSectionResults: FC<ExpandableSectionResultsProps> = ({
   );
   const analysisType =
     jobConfig && jobConfig.analysis ? getAnalysisType(jobConfig.analysis) : undefined;
+
+  const generateDiscoverUrl = async (rowIndex: number) => {
+    const item = tableItems[rowIndex];
+
+    if (discoverLocator !== undefined) {
+      const url = await discoverLocator.getRedirectUrl({
+        indexPatternId: dataViewId,
+        timeRange: data.query.timefilter.timefilter.getTime(),
+        filters: data.query.filterManager.getFilters(),
+        query: {
+          language: SEARCH_QUERY_LANGUAGE.KUERY,
+          // Filter for all visible column values except the results field values
+          query: indexData.visibleColumns
+            .filter((column) => !column.includes(resultsField!))
+            .map((column) => {
+              if (item[column] !== undefined) {
```

Comment:
if `item[column]` is `undefined` it looks like this string will be incorrect as `undefined` will be added to it.
Maybe there should be a filter before the map to remove this risk?

Reply:
Oh yes! Good catch - I'll add it to the filter above 👍